misc-scripts/xref_mapping/XrefParser/BaseParser.pm

```diff
@@ -1133,30 +1130,66 @@ ADX
     ) or croak("$acc\t$label\t\t$source_id\t$species_id\n");
   }
   $add_xref_sth->finish();
-  $dependent_id = $self->get_xref($acc, $source_id, $species_id, $dbi);
 
   ################################################
-  # Croak if we have failed to create.get the xref
+  # Croak if we have failed to create/get the xref
   ################################################
-  if(!(defined $dependent_id)){
+  $dependent_id = $self->get_xref($acc, $source_id, $species_id, $dbi);
+  if ( !(defined $dependent_id) ) {
     croak("$acc\t$label\t\t$source_id\t$species_id\n");
   }
 
-  ########################################################################################
-  # If the dependency has not already been set ( is already in hash xref_dependent_mapped)
-  # then add it
-  ########################################################################################
-  if(!(defined $xref_dependent_mapped{"$master_xref|$dependent_id"}) || $xref_dependent_mapped{"$master_xref|$dependent_id"} ne $linkage){
-    $add_dependent_xref_sth->execute( $master_xref, $dependent_id, $linkage,
-				      $source_id )
-      or croak("$master_xref\t$dependent_id\t$linkage\t$source_id");
-    $xref_dependent_mapped{"$master_xref|$dependent_id"} = $linkage;
+  ################################
+  # Now add the dependency mapping
+  ################################
+  $self->add_dependent_xref_maponly( $dependent_id, $source_id,
+                                     $master_xref, $linkage,
+                                     $dbi );
+
+  return $dependent_id;
+}
+
+
+##################################################################
+# Add a single record to the dependent_xref table.
+# Note that an xref must already have been added to the xref table
+# Note that a corresponding method for direct xrefs is called add_direct_xref()
+##################################################################
+
+sub add_dependent_xref_maponly {
```

Comment:
Naming is a bit strange, all of it...
add_direct_xref links in function to add_dependent_xref_maponly
add_to_direct_xrefs links in function to add_dependent_xref

I'm sure we could make it more clear, maybe in the design refactoring.

Reply:
Agree - add_dependent_xref() would be more meaningfully named add_xref_and_dependency(), and add_dependent_xref_maponly() maybe set_dependency_for_xref()

but calls to add_dependent_xref() are already everywhere in the parsers...

Reply:
Indeed, the naming is very much due for refactoring.